### PR TITLE
Adding back Principal check and removing checks on multiple conditions

### PR DESCRIPTION
**Pull request by jhnlsn**
This addresses issue #1894

### CODE REVIEW

**Comment by kapilt** [comment]
simply removing the asserts is not sufficient, the logic that necessitated those asserts needs to be updated as well.

**Comment by jhnlsn** [comment]
There wasn't any original logic around those asserts, which is why they are being removed.

**Comment by jhnlsn** [comment]
@kapilt if you can shed some more light as to why those asserts were added, even though multiple conditions are allowed as per notes found by kit in the issue noted in this PR that would be helpful.

**Comment by jhnlsn** [comment]
I reviewed handle_condition with the original and they are doing things the same by always getting the first item off the conditions array.  The original did not throw when more than one item in the array existed.  We need to remove these asserts so that we can move forward with this code and then if we need to handle multiple conditions, we can do that later.  Ultimately there is no reason to throw on multiple conditions.

File: c7n/filters/iamaccess.py [hunk 1]
```diff
@@ -152,23 +154,34 @@ def handle_principal(self, s):
...
-        if key in self.whitelist_conditions:
+        if c['key'] in self.whitelist_conditions:
             return True
-        handler_name = "handle_%s" % key.replace('-', '_').replace(':', '_')
+        handler_name = "handle_%s" % c['key'].replace('-', '_').replace(':', '_')
         handler = getattr(self, handler_name, None)
         if handler is None:
-            print("no handler:%s op:%s key:%s value:%s" % (
-                handler_name, op, key, value))
+            print("no handler:%s op:%s key:%s values:%s" % (
```
Comment (kapilt) [inline c7n/filters/iamaccess.py]: i realize this was here before but could you change this to a log.warning

**Pull request merged**